pennylane/_qubit_device.py

```diff
@@ -781,36 +798,78 @@ def estimate_probability(self, wires=None, shot_range=None, bin_size=None):
         """
 
         wires = wires or self.wires
-        # convert to a wires object
+        # convert to a Wires object
         wires = Wires(wires)
         # translate to wire labels used by device
         device_wires = self.map_wires(wires)
+        num_wires = len(device_wires)
 
-        sample_slice = Ellipsis if shot_range is None else slice(*shot_range)
-        samples = self._samples[sample_slice, device_wires]
+        if shot_range is None:
+            samples = self._samples[..., device_wires]
+        else:
+            samples = self._samples[..., slice(*shot_range), device_wires]
```

Comment:
In the first case, `...` corresponds to the broadcasting and shots dimensions, or only to the shots dimension.
In the second case, it corresponds to the broadcasting dimensions, or to nothing.